File: deploy.js
require('dotenv').config();
const { REST } = require('@discordjs/rest');
const { Routes } = require('discord-api-types/v9');
const { SlashCommandBuilder } = require('@discordjs/builders');

const commands = [
  new SlashCommandBuilder()
    .setName('tournamentqueue')
    .setDescription('Create a tournament queue')
    .addStringOption(option =>
      option.setName('name')
        .setDescription('Tournament name')
        .setRequired(true))
    .addStringOption(option =>
      option.setName('timestamp')
        .setDescription('Timestamp for join deadline (unix)')
        .setRequired(true)),
];

const rest = new REST({ version: '9' }).setToken(process.env.DISCORD_TOKEN);

(async () => {
  try {
    console.log('Started refreshing application (/) commands.');

    await rest.put(
      Routes.applicationGuildCommands(process.env.CLIENT_ID, process.env.GUILD_ID),
      { body: commands },
    );

    console.log('Successfully reloaded application (/) commands.');
  } catch (error) {
    console.error(error);
  }
})();

File: deploy.py
import os
import asyncio
from dotenv import load_dotenv

import discord

load_dotenv()

GUILD_ID = int(os.getenv('GUILD_ID'))
DISCORD_TOKEN = os.getenv('DISCORD_TOKEN')

# Import bot from index.py (with commands already defined)
from index import bot

async def deploy_commands():
    """Deploy slash commands to Discord."""
    GUILD_ID = int(os.getenv('GUILD_ID'))
    await bot.tree.sync(guild=discord.Object(id=GUILD_ID))
    print('Successfully reloaded application (/) commands.')
    await bot.close()

if __name__ == '__main__':
    asyncio.run(deploy_commands())
